scripts/HighNoon.js

```diff
@@ -0,0 +1,401 @@
+var TriggerHelper = require('./lib/TriggerHelper.js');
+var connectionHelper = require('./lib/ConnectionHelper.js');
+var help = require('./lib/Help.js');
+var messageHelper = require('./lib/MessageHelper.js');
+
+help.setHelpCategory(
+  'highnoon',
+  'mcree\'s favorite time of the day',
+  '/highnoon test - hello world message\n' +
+  '/highnoon - use mcree\'s ultimate\n' +
+  '/highnoon spin - spins the barrel\n' +
+  '/highnoon stats - display the game scores'
+);
+
+var Highnoon = function (robot) {
+  var trigger = new TriggerHelper('highnoon');
+  var sessionStore = {};
+
+  ////////////// STATES //////////////
+
+  const PLAYER_ADDED = 'PLAYER_ADDED';
+  const PLAYER_REMOVED = 'PLAYER_REMOVED';
+  const PLAYER_ALREADY_IN_SESSION = 'PLAYER_ALREADY_IN_SESSION';
+  const PLAYER_DOES_NOT_EXIST = 'PLAYER_DOES_NOT_EXIST';
+  const SESSION_CREATED = 'SESSION_CREATED';
+  const SESSION_FULL = 'SESSION_FULL';
+
+
+  ////////////// CLASSES //////////////
+
+  class PlayerState {
+    constructor(name) {
+      this.name = name;
+      this.alive = true;
+    }
+
+    getName() {
+      return this.name;
+    }
+
+    isAlive() {
+      return this.alive;
+    }
+
+    gotShot() {
+      this.alive = false;
+    }
+  };
+
+  class GameState {
+    constructor(maxPlayers) {
+      this.players = {};
+      this.maxPlayers = maxPlayers;
+      this.shotsFired = false;
+      this.countdownValue = Math.floor(Math.random() * 2500) + 2500; // Guarantees to never be zero
+      this.timer = null;
+      this.elapsed = 0;
+    }
+
+    playersNeeded() {
+      return this.maxPlayers - this.players.length;
+    }
+
+    getCurrentPlayers() {
+      return this.players;
+    }
+
+    getMaxPlayers() {
+      return this.maxPlayers;
+    }
+
+    getCountDownValue() {
+      return this.countdownValue;
+    }
+
+    sessionIsFull() {
+      return this.players.length === this.maxPlayers;
+    }
+
+    incrementTimer() {
+      this.elapsed += 2500;
+    }
+
+    timeToDraw() {
+      return this.elapsed >= this.countDownValue;
+    }
+
+    duelIsDone() {
+      this.allShotsFired = true;
+    }
+
+  };
+
+  class Session {
```

Comment:
Minor naming nitpick:  This `Session` class behaves more like your `GameEngine`, rather than a "`Session`" (which implies its a stateful, temporal thing).